impl/src/metadata.rs

```diff
@@ -12,34 +12,21 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use cargo::{
-  core::{
-    dependency::Kind as CargoKind, resolver::ResolveOpts,
-    summary::FeatureValue as CargoFeatureValue, Workspace,
-  },
-  ops::{self, Packages},
-  util::Config,
-  CargoResult,
-};
+use cargo::CargoResult;
 
-use serde_json;
-use std::{collections::HashMap, env, fs, path::PathBuf, process::Command};
+use cargo_metadata::MetadataCommand;
+pub use cargo_metadata::{DependencyKind, Metadata, Node, Package, PackageId};
 
-use crate::util::{self, RazeError};
-use tempdir::TempDir;
+use std::{fs, path::PathBuf};
 
-use serde_derive::{Deserialize, Serialize};
+use tempdir::TempDir;
 
 const SYSTEM_CARGO_BIN_PATH: &str = "cargo";
 
-pub type PackageId = String;
-pub type Kind = String;
-pub type TargetSpec = String;
-
 /**
  * An entity that can retrive deserialized metadata for a Cargo Workspace.
  *
- * The `CargoInternalsMetadataFetcher` is probably the one you want.
+ * The `CargoSubcommandMetadataFetcher` is probably the one you want.
```

Comment:
It's probably safe to just drop this comment. It was more cautionary before when there was one "good" implementation and one "experimental" implementation.

(no reason to annotate every trait with the preferred implementation of the trait, in general)

Reply:
Done!